pkgs/jnigen/lib/src/bindings/renamer.dart

```diff
@@ -227,7 +235,8 @@ class _MethodRenamer implements Visitor<Method, void> {
 
   @override
   void visit(Method node) {
-    final name = _preprocess(node.isConstructor ? 'new' : node.name);
+    final name = _preprocess(
+        node.isConstructor ? 'new' : node.userDefinedName ?? node.name);
```

Comment:
The user should technically be able to rename the constructors as well, I'm not sure that we currently support this in the dart_generator though. So let's fix that in another PR.